src/preact/base-element.js

```diff
@@ -878,6 +875,13 @@ PreactBaseElement['usesTemplate'] = false;
  */
 PreactBaseElement['shadowCss'] = null;
 
+/**
+ * Whether this element uses Shadow DOM.
+ *
+ * @protected {boolean}
+ */
+PreactBaseElement['usesShadowDom'] = !!PreactBaseElement['shadowCss'];
```

Comment:
I think it's cleaner to just go with the constant here. Otherwise someone might get the wrong impression that only overriding `shadowCss` would change this value.